src/generate-types.ts

```diff
@@ -791,6 +793,53 @@ export function run(options: Options) {
     return ts.createIdentifier(prop);
   }
 
+  function generateDiscriminatorMapping(
+    discriminatorField: string,
+    schemas: readonly SchemaObject[]
+  ): [ts.Expression, (ts.Identifier | ts.PropertyAccessExpression)[]] {
+    const [refs, internals] = _.partition(
+      schemas,
+      (schema): schema is ReferenceObject => '$ref' in schema
+    );
+    const properties = _.flatMap(internals, schema => {
+      if (schema.type !== 'object') {
+        throw new Error('schemas in oneOf with discriminator need to be objects');
+      }
+      let discriminatorFieldSchema;
+      if (
+        !schema.properties ||
+        !(discriminatorFieldSchema = schema.properties[discriminatorField])
+      ) {
+        throw new Error(`field ${discriminatorField} not defined, required by discriminator`);
+      }
+      if (!('type' in discriminatorFieldSchema) || discriminatorFieldSchema.type !== 'string') {
+        throw new Error('Only string type supported type supported');
+      }
+      if (!('enum' in discriminatorFieldSchema) || !discriminatorFieldSchema.enum) {
+        throw new Error('Must have enum');
```

Comment:
why do we need `enum`? 🤔

Reply:
If you don't supply manually created mappings, we try to extract the mappings from the types automatically. This works by going through the discriminator field in all the options, and picking up enum values from there.

Reply:
OHH right! cool 😁

Reply:
So with:

```
oneOf:
  - type: object
    properties: 
      type:
        type: string
        enum: [ value1 ]
  - type: object
    properties: 
      type:
        type: string
        enum: [ value2, value3 ]
discriminator:
  propertyName: type
```

We'll redirect value1 to first type, and value2 and value3 to second type.